src/coreclr/tools/aot/ILCompiler.ReadyToRun/ObjectWriter/MapFileBuilder.cs

```diff
@@ -205,6 +226,20 @@ public void SaveCsv(string nodeStatsCsvFileName, string mapCsvFileName)
             }
         }
 
+        public void SavePdb(string pdbPath, string dllFileName)
```

Comment:
It feels weird having `MapFileBuilder` be the thing that writes PDBs and Perf maps since its purpose is to write a text summary file at the end of compilation. The map file builder's data structures at this point are a nice model of node to file location which makes it handy for these purposes. Maybe we could split the concepts into a class containing mapping details (shareable by map, perfmap, pdb generation)?

Reply:
Fixed in 9th commit - I have split the logic into an "output file info collector" (OutputInfoBuilder) and the actual emitters using it - MapFileBuilder and SymbolFileBuilder, thanks for the suggestion.